trace/rate_limit_test.go

```diff
@@ -44,7 +44,7 @@ func TestRateLimit_FirstSample(t *testing.T) {
 	start := time.Now()
 	cc := &countingCollector{}
 	rl := NewRateLimit(1.0)
-	c := rl.NewCollector(cc).(*rateLimitCollector)
+	c := rl.NewCollector(cc, NewPacketCounter()).(*rateLimitCollector)
```

Comment:
Non-blocking: Might be a good idea for this unit test to check the value in the PacketCounter afterwards.  There should be 5 dropped packets.  But I'm not sure the test successfully waits for all of the drops to occur?  (It only waits for successes.)